### ISSUE TITLE
Changing Sheduler during training

### ISSUE THREAD

**Issue by 0phoff** [PyTorchLightning/pytorch-lightning#3070]
## 🚀 Feature
I would like to be able to change schedulers during training:
  - Warm-up learning rate for the first X batches (eg. 1000)
  - Other learning rate (ag. ReduceLROnPlateau, StepLR, etc.)

### Motivation
I would like to have a way to switch schedulers for a certain optimizer during training.  
I think it is fairly common to have a warm-up learning rate in order to avoid exploding loss values early on in the training, but this should not force us to keep that learning rate schedule for the entirety of our training.

### Alternatives
You could probably implement everything inside of a LambdaLR scheduler, but this seems like a hack and is not entirely possible if you want to combine eg. ReduceLROnPlateau with another scheduler.

### Additional context
I implemented a [SchedulerCompositor](https://gitlab.com/EAVISE/lightnet/-/blob/master/lightnet/engine/_scheduler.py) in my own library, but it still has a few flaws:
 - You need to pass in a `count` argument to the `step()` call, which means it is not really possible to transparently swap out a normal scheduler for my compositor.
 - Ideally, you would like to be able to step one scheduler after each batch, and the next one after each epoch.

The way I see it, there are 2 ways to implement this.  
You can either implement some kind of scheduler-compositor, which wraps your schedulers and calls the correct one depending on which count you are (this is what I did in lightnet).  
Otherwise, you could also return multiple schedulers for one optimizer in `configure_optimizers()`, and the logic for choosing a scheduler could be implemented straight in the training loop.
